airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StatusView.tsx

```diff
@@ -74,56 +82,70 @@ const StatusView: React.FC<StatusViewProps> = ({ connection, isStatusUpdating })
-    >
-      {showFeedback ? (
-        <FormattedMessage id="sources.syncingNow" />
-      ) : (
-        <>
-          <TryArrow icon={faRedoAlt} />
-          <FormattedMessage id="sources.syncNow" />
-        </>
-      )}
-    </SyncButton>
+  const onCancelJob = () => {
+    if (!activeJob?.id) {
+      return;
+    }
+    setActiveJob((state) => ({ ...state, isCanceling: true } as ActiveJob));
+    return cancelJob(activeJob.id);
+  };
+
+  const cancelJobBtn = (
+    <Button className={styles.cancelButton} disabled={!activeJob?.id || activeJob.isCanceling} onClick={onCancelJob}>
+      <FontAwesomeIcon className={styles.iconXmark} icon={faXmark} />
+      {activeJob?.action === ActionType.RESET && <FormattedMessage id="connection.cancelReset" />}
+      {activeJob?.action === ActionType.SYNC && <FormattedMessage id="connection.cancelSync" />}
+    </Button>
   );
 
   return (
-    <StyledContentCard
-      title={
-        <Title>
-          <FormattedMessage id="sources.syncHistory" />
-          {connection.status === ConnectionStatus.active && (
-            <div>
-              <ToolTip control={resetDataBtn} disabled={!isAtLeastOneJobRunningOrPending} cursor="not-allowed">
-                <FormattedMessage id="connection.pendingSync" />
-              </ToolTip>
-              <ToolTip control={syncNowBtn} disabled={!isAtLeastOneJobRunningOrPending} cursor="not-allowed">
-                <FormattedMessage id="connection.pendingSync" />
-              </ToolTip>
-            </div>
-          )}
-        </Title>
-      }
-    >
-      {jobs.length ? <JobsList jobs={jobs} /> : <EmptyResource text={<FormattedMessage id="sources.noSync" />} />}
-    </StyledContentCard>
+    <div className={styles.statusView}>
+      <ContentCard
+        className={styles.contentCard}
+        title={
+          <div className={styles.title}>
+            <FormattedMessage id="sources.syncHistory" />
+            {connection.status === ConnectionStatus.active && (
+              <div className={styles.actions}>
+                {!activeJob?.action && (
+                  <>
+                    <Button className={styles.resetButton} secondary onClick={onResetDataButtonClick}>
+                      <FormattedMessage id="connection.resetData" />
+                    </Button>
+                    <Button className={styles.syncButton} disabled={!allowSync} onClick={onSyncNowButtonClick}>
+                      <div className={styles.iconRotate}>
+                        <RotateIcon />
+                      </div>
+                      <FormattedMessage id="sources.syncNow" />
+                    </Button>
+                  </>
+                )}
+                {activeJob?.action && !activeJob.isCanceling && cancelJobBtn}
+                {activeJob?.action && activeJob.isCanceling && (
+                  <ToolTip control={cancelJobBtn} cursor="not-allowed">
+                    <FormattedMessage id="form.canceling" />
```

Comment:
The value of `form.canceling` should be `Canceling...` instead of `Canceling`

Reply:
Fixed. Thanks